kafka-client/src/main/java/dev/responsive/kafka/api/ResponsiveWindowedStoreSupplier.java

```diff
@@ -17,66 +17,48 @@
 package dev.responsive.kafka.api;
 
 import dev.responsive.kafka.store.ResponsiveWindowStore;
-import dev.responsive.utils.TableName;
 import org.apache.kafka.common.utils.Bytes;
 import org.apache.kafka.streams.state.WindowBytesStoreSupplier;
 import org.apache.kafka.streams.state.WindowStore;
 
 public class ResponsiveWindowedStoreSupplier implements WindowBytesStoreSupplier {
 
-  private final TableName name;
-  private final long retentionPeriod;
-  private final long windowSize;
-  private final boolean retainDuplicates;
+  private final ResponsiveWindowParams params;
+  private final long segmentIntervalMs;
 
-  public ResponsiveWindowedStoreSupplier(
-      final String name,
-      final long retentionPeriod,
-      final long windowSize,
-      final boolean retainDuplicates
-  ) {
-    this.name = new TableName(name);
-    this.retentionPeriod = retentionPeriod;
-    this.windowSize = windowSize;
-    this.retainDuplicates = retainDuplicates;
+  public ResponsiveWindowedStoreSupplier(final ResponsiveWindowParams params) {
+    this.params = params;
+    this.segmentIntervalMs = params.retentionPeriod() / params.numSegments();
   }
 
   @Override
   public String name() {
-    return name.kafkaName();
+    return params.name().kafkaName();
   }
 
   @Override
   public WindowStore<Bytes, byte[]> get() {
-    return new ResponsiveWindowStore(
-        name,
-        retentionPeriod,
-        windowSize,
-        retainDuplicates
-    );
+    return new ResponsiveWindowStore(params);
   }
 
-  // Responsive window store is not *really* segmented, so just
-  // say size is 1 ms this is what InMemoryWindowBytesStoreSupplier
-  // does in Kafka Streams
   @Override
   public long segmentIntervalMs() {
-    return 1;
+    return segmentIntervalMs;
```

Comment:
I looked it up, seems like it's mainly/only used these days for the cache: basically in the Caching layer we prefix the key (technically combined key + windowStartTime) with the segmentId so that it follows the pattern of the underling state store.

I believe it would mess with the ordering of key & time range scans and possibly even result in duplicates or skipped records. The way range scans work is to create two iterators: one on the cache and one on the underlying store. We keep track of the next element for each iterator, and when the user calls `next()` we compare these two and return the one that comes first (bytes wise). This really only works when the cache is segmented/formatted in the same way as the data store itself

(It's also used for that optimization I mentioned that I wrote years ago: where we skip over the rest of the segment and only scan the part we're interested in for transegmental (probably not a word but I like it) range scans)